mylang/extensions.bzl

```diff
@@ -0,0 +1,35 @@
+"extensions for bzlmod"
+
+load(":repositories.bzl", "mylang_register_toolchains")
+
+mylang_toolchain = tag_class(attrs = {
+    "name": attr.string(doc = "Base name for generated repositories"),
+    "mylang_version": attr.string(doc = "Explicit version of mylang."),
+})
+
+def _toolchain_extension(module_ctx):
+    registrations = {}
+    for mod in module_ctx.modules:
+        for toolchain in mod.tags.toolchain:
+            if toolchain.name in registrations.keys():
+                if toolchain.mylang_version == registrations[toolchain.name]:
+                    # No problem to register a matching toolchain twice
+                    continue
+                fail("Multiple conflicting toolchains declared for name {} ({} and {}".format(
+                    toolchain.name,
+                    toolchain.mylang_version,
+                    registrations[toolchain.name],
+                ))
+            else:
+                registrations[toolchain.name] = toolchain.mylang_version
+    for name, mylang_version in registrations.items():
+        mylang_register_toolchains(
+            name = name,
```

Comment:
FYI https://github.com/bazel-contrib/rules_cuda/pull/66#discussion_r1096604845 - we may not want this pattern in rules-template..

Reply:
@alexeagle @Wyverald Should we (and potentially others) discuss and codify some best practices around module extensions? Could be an agenda point for the next SIG meeting.

Reply:
good point! yeah we shouldn't use a "name" parameter if we expect it to be unique only. rules_jvm_external is fine since it just aggregates tags with the same name.

Reply:
@fmeum PTAL - I've changed this to be named "mylang" where only the root module can declare other names.

Reply:
I added a similar fix here for the node toolchain: https://github.com/bazelbuild/rules_nodejs/pull/3626

Reply:
IIUC the result is that every module can define a toolchain version under the default name, the latest of those versions will be selected (the rest discarded), and will always be registered by `rules_mylang`. Additionally, the root module can define arbitrarily many more toolchain versions under different names (the latest version will be picked for each name) and can register them as it sees fit, effectively overriding the default named toolchain due to toolchain resolution precedence.

That sounds like a reasonable starting point to me. IIUC the added complexity in rules_go comes from the fact that they offer config settings to switch between Go SDK versions as well.

Reply:
thanks for summarizing, added that to the docstring for this module (ahem. "starlark module") 😁